### PR TITLE
Add libdeflate and libxsmm

### PR DESCRIPTION
**Pull request by bgruening**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bgruening** [comment]
blocked by https://github.com/conda-forge/staged-recipes/pull/9295

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -268,6 +270,8 @@ pin_run_as_build:
     max_pin: x.x
   libuuid:
     max_pin: x
+  libxsmm:
+    max_pin: x.x
```
[the remark this answers is not in the document]
Comment (bgruening) [reply recipe/conda_build_config.yaml]: Mh, than we need to fix that in the recipe.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -228,6 +228,8 @@ pin_run_as_build:
     max_pin: x.x
   libcurl:
     max_pin: x
+  libdeflate:
+    max_pin: x.x
```
[the remark this answers is not in the document]
Comment (bgruening) [reply recipe/conda_build_config.yaml]: fixed.

**Comment by scopatz** [comment]
Do you want to update the date/version?

**Comment by CJ-Wright** [comment]
Would it be possible to do this via CFEP9?

File: recipe/conda_build_config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipe/conda_build_config.yaml]: In the recipe it is `x`, should we change it to x.x or should we change it here to x?

See https://github.com/conda-forge/libxsmm-feedstock/blob/master/recipe/meta.yaml#L16

**Comment by hmaarrfk** [comment]
We should use a migrator for this.

**Comment by hmaarrfk** [comment]
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2510
and
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2511

**Comment by hmaarrfk** [comment]
Sorry for letting this one linger.

**Pull request closed without merging**